pkg/operator/resource/resourceapply/monitoring.go

```diff
@@ -136,9 +136,13 @@ func ApplyUnstructuredResourceImproved(
 		return nil, false, err
 	}
 
-	// Check if the metadata objects differ.
+	// Check if the metadata objects differ. This only checks for selective fields (excluding the resource version, among others).
 	didMetadataModify := ptr.To(false)
 	resourcemerge.EnsureObjectMeta(didMetadataModify, &existingObjectMetaTyped, requiredObjectMetaTyped)
+	existingCopy.Object["metadata"], err = runtime.DefaultUnstructuredConverter.ToUnstructured(&existingObjectMetaTyped)
```

Comment:
shouldn't we update the metadata conditionally  ? (when didMetadataModify == true)

Reply:
ACK, done.